### ISSUE TITLE
Verify lifecycle policies

### ISSUE THREAD

**Issue by pew** [minio/minio#10980]
## Expected Behavior

I expect that lifecycle policies applied to a bucket will run and be executed.

## Current Behavior

Files will just stay inside the bucket, nothing will be deleted. This is my lifecycle policy applied to a bucket:

```
{
	"Rules": [{
		"Expiration": {
			"Days": 1
		},
		"ID": "delete after 1 day",
		"Filter": {
			"Prefix": ""
		},
		"Status": "Enabled"
	}]
}
```

I'd expect that files are being deleted after 1 day, I've read in some other issues that lifecycle policies will kick in after 24h of running minio, but I've got files in there which are multiple weeks old. The policy was applied right after bucket creation without any files in there.

## Your Environment

I'm running the [minio kubernetes helm chart](https://github.com/minio/charts) with the image `minio/minio:RELEASE.2020-11-06T23-17-07Z` and the only modification I provided with helm was setting the environment variable `MINIO_BROWSER: "off"`

**Comment by pew** [comment]
When I execute `mc admin info` I get something along these lines:

```
   Uptime: 1 week 
   Version: 2020-11-06T23:17:07Z
   Network: 1/1 OK 
18 GiB Used, 6 Buckets, 1,821,778 Objects
```

Does this mean minio thinks that there are only 18 GiB in use across 6 buckets but it's somehow taking up 1,6TB on my actual disk and the files are still in the individual folders when I check on the server itself

**Comment by vadmeste** [comment]
@pew can you run this command on one of the object that you expect to be removed with the lifecycle expiry ?

`mc stat alias/bucket/path/to/your/object`

**Comment by pem** [comment]
> @pem can you run this command on one of the object that you expect to be removed with the lifecycle expiry ?
> 
> `mc stat alias/bucket/path/to/your/object`

@vadmeste I think you tagged the wrong person.

**Comment by vadmeste** [comment]
> @vadmeste I think you tagged the wrong person.

Thanks, fixed. Sorry for disturbance.

**Comment by pew** [comment]
```
Name      : 1.png
Date      : 2020-11-18 07:45:10 UTC 
Size      : 2.5 KiB 
ETag      : b54007fcaf6c59ab61d521eb7c4e8318 
Type      : file 
Expiration: 2020-11-20 00:00:00 UTC (lifecycle-rule-id: delete after 1 day) 
Metadata  :
  Content-Type: application/octet-stream 
```

**Comment by josiasmontag** [comment]
Same problem here. Policy:
```
{
    "Rules": [
        {
           "Filter": {
                "Prefix": ""
            },
            "Expiration": {
                "Days": 1
            },
            "ID": "DeleteDaily",
            "Status": "Enabled",
            "AbortIncompleteMultipartUpload": {
        		"DaysAfterInitiation": 1
      		}
        }
    ]
}
```

```
➜  ~ mc stat minio/test/test/test.txt 
Name      : test.txt
Date      : 2020-11-26 15:44:38 CET 
Size      : 4 B    
ETag      : 098f6bcd4621d373cade4e832627b4f6 
Type      : file 
Expiration: 2020-11-28 01:00:00 CET (lifecycle-rule-id: DeleteDaily) 
Metadata  :
  Content-Type: application/octet-stream 
```


MinIO version `2020-10-28`, Docker Swarm

**Comment by vadmeste** [comment]
@pew I am not able to reproduce the issue. Is data usage still wrong ? Could you set `MINIO_DISK_USAGE_CRAWL_DEBUG` to `on` in your environment ? It will print more logs about the crawler and we can check if the crawler is crawling the bucket in question.

**Comment by pew** [comment]
@vadmeste files are still not being deleted, as a workaround I added a cronjob to run the `mc` client manually with the flag `--older-than`

I added the environment variable and restarted the process, can you help me out what I'm looking for in the log files now? This is the output right now:

```
dataUpdateTracker: Marking path dirty: bucketname/6ae268ec-782f-4351-8fb9-328f75e97e2c
dataUpdateTracker: Marking path dirty: bucketname/6ae268ec-782f-4351-8fb9-328f75e97e2c/8
dataUpdateTracker: Marking path dirty: bucketname/6ae268ec-782f-4351-8fb9-328f75e97e2c/8/137
dataUpdateTracker: Marking path dirty: bucketname/6ae268ec-782f-4351-8fb9-328f75e97e2c/8/137/169.png
```

`mc stat alias/bucketname/6ae268ec-782f-4351-8fb9-328f75e97e2c/8/137/169.png`:

```
Name      : 169.png
Date      : 2020-12-02 03:02:26 UTC 
Size      : 30 KiB 
ETag      : 2a7822feaa40347a8c53a8aa0fe4cc96 
Type      : file 
Expiration: 2020-12-04 00:00:00 UTC (lifecycle-rule-id: delete after 1 day) 
Metadata  :
  Content-Type: application/octet-stream 
```

**Comment by vadmeste** [comment]
@pew something like 

```data-usage:  Crawl time: 927.221µs -  /path/to/ + your-bucket-name```

**Comment by pew** [comment]
thank you @vadmeste - just checked the logs since I restarted:

```
crawlBucket: lifecycle: Active rules found
data-usage: Start crawling. Bloom filter: false - /minio + bucketB
data-usage: Cycle: 8, Entries: 34876 - /minio + bucketB
data-usage: Level 0, scanning 1 directories. - /minio + bucketB
folder-scanner: Prefix "" has active rules
dataUpdateTracker: Marking path dirty: bucketB
dataUpdateTracker: Marking path dirty: bucketB/001b43bb-4a01-4c47-b38c-01bbee68981c
dataUpdateTracker: Marking path dirty: bucketB/001b43bb-4a01-4c47-b38c-01bbee68981c/8
dataUpdateTracker: Marking path dirty: bucketB/001b43bb-4a01-4c47-b38c-01bbee68981c/8/134
dataUpdateTracker: Marking path dirty: bucketB/001b43bb-4a01-4c47-b38c-01bbee68981c/8/134/170.png
data-usage: Level 1, scanning 176877 directories. - /minio + bucketB
folder-scanner: Prefix "f6f78b58-c914-4253-98be-613516727632" has active rules
folder-scanner: Prefix "e801d0ec-9d04-4786-bfa9-135a9b0e2528" has active rules
folder-scanner: Prefix "c7fd0f63-cddc-4a4f-a794-8a1ca9b58e9e" has active rules
data-usage: New folders: [{bucketB/3f5c5780-978f-4ad6-beeb-d1c7548b2a54/8 0xc002e500c0 1} {bucketB/3f5c5780-978f-4ad6-beeb-d1c7548b2a54/7 0xc002e500c0 1} {bucketB/3f5c5780-978f-4ad6-beeb-d1c7548b2a54/6 0xc002e500c0 1} {bucketB/3f5c5780-978f-4ad6-beeb-d1c7548b2a54/4 0xc002e500c0 1} {bucketB/3f5c5780-978f-4ad6-beeb-d1c7548b2a54/3 0xc002e500c0 1} {bucketB/3f5c5780-978f-4ad6-beeb-d1c7548b2a54/2 0xc002e500c0 1} {bucketB/3f5c5780-978f-4ad6-beeb-d1c7548b2a54/5 0xc002e500c0 1} 
```

the last line with *New folders* goes on for a very, very long line printing all the folders

**Comment by vadmeste** [comment]
@pew does lifecycle work for bucketB ?

**Comment by pew** [comment]
No, lifecycle policies don't work at all for me across all buckets:

```
Name      : 173.png
Date      : 2020-11-14 03:34:17 UTC 
Size      : 5.5 KiB 
ETag      : fd4f507ba1032353b7a3dea01c8b2749 
Type      : file 
Expiration: 2020-11-16 00:00:00 UTC (lifecycle-rule-id: delete after 1 day) 
Metadata  :
  Content-Type: application/octet-stream 
```

**Comment by vadmeste** [comment]
> No, lifecycle policies don't work at all for me across all buckets:

could you please confirm that 173.png belongs to bucketB ? I just need to know if the crawling is working for that bucket or not.

**Comment by pew** [comment]
This file was deleted manually in the meantime. I just did the whole thing with another file in the same bucket again, so I can confirm that the file belongs to the bucket:

```
data-usage: Start crawling. Bloom filter: false - /minio + bucketA
data-usage: Cycle: 8, Entries: 1 - /minio + bucketA
data-usage: Level 0, scanning 1 directories. - /minio + bucketA
data-usage: Level 1, scanning 0 directories. - /minio + bucketA
data-usage: New folders: [] - /minio + bucketA
data-usage: Existing folders: 0 - /minio + bucketA
data-usage: Finished crawl, 1 entries - /minio + bucketA
data-usage:  Crawl time: 11.183083ms - /minio + bucketA
data-usage: Start crawling. Bloom filter: false - /minio + bucketB
data-usage: Cycle: 8, Entries: 34876 - /minio + bucketB
data-usage: Level 0, scanning 1 directories. - /minio + bucketB
data-usage: Level 1, scanning 176877 directories. - /minio + bucketB
```

`mc stat alias/bucketB/c7fd0f63-cddc-4a4f-a794-8a1ca9b58e9e/8/142/178.png`

```
Name      : 178.png
Date      : 2020-11-14 03:37:21 UTC 
Size      : 2.1 KiB 
ETag      : 615c48adb3621139910c50a53aad85b7 
Type      : file 
Expiration: 2020-11-16 00:00:00 UTC (lifecycle-rule-id: delete after 1 day) 
Metadata  :
  Content-Type: application/octet-stream 
```

**Comment by vadmeste** [comment]
@pew do you ever see `data-usage:  Crawl time: xxxx - /minio + bucketB` ? it might take while if bucketB contains a lot of objects

**Comment by pew** [comment]
@vadmeste No, that's not showing up in the log files. Only for one other bucket with basically no files in it. When I did add the environment variable I restarted the server, so it's been running for ~35 hours now. Before that it was running for a few weeks, though.

**Comment by vadmeste** [comment]
> @vadmeste No, that's not showing up in the log files. Only for one other bucket with basically no files in it. When I did add the environment variable I restarted the server, so it's been running for ~35 hours now. Before that it was running for a few weeks, though.

Did you ever upgrade your server from old versions ?

**Comment by pew** [comment]
@vadmeste I only upgraded the helm chart once from `RELEASE.2020-10-18T21-54-12Z` to `RELEASE.2020-11-06T23-17-07Z`

**Comment by cubbasa** [comment]
@pew Do You have intensive traffic on this setup (like large volume of data get in and expired soon because based on previous comments seem so)?
@vadmeste seems that I've got similar situation that I've a lot of data coming in (few TB of data of small files) and expiring them after 7 days. Minio crawler is too slow to delete all expired files on time and leave a lot of expired objects behind which increase volume usage by a lot each day. To avoid this situation I've got extra cronjob with minio client which delete all expired objects with flag --older-than to support Minio crawler.

**Comment by harshavardhana** [comment]
> @vadmeste seems that I've got similar situation that I've a lot of data coming in (few TB of data of small files) and expiring them after 7 days. Minio crawler is too slow to delete all expired files on time and leave a lot of expired objects behind which increase volume usage by a lot each day. To avoid this situation I've got extra cronjob with minio client which delete all expired objects with flag --older-than to support Minio crawler.

@cubbasa if your volume is high just set the `MINIO_DISK_USAGE_CRAWL_DELAY=0` this will run the lifecycle at full speed.

**Comment by pew** [comment]
@cubbasa Yes, there's a ton of traffic coming in with a lot of small files and some of them expire after a day already. Right now the cronjob I've got in place is running fine and it's deleting things. We had the cronjobs in place before the lifecycle policies, I just wanted to remove them and use the native functionality.

@harshavardhana  I'm going to try to set the `MINIO_DISK_USAGE_CRAWL_DELAY=0` environment variable, disable the cronjobs and report back.

**Comment by pew** [comment]
Now I've been running minio for a while with the `MINIO_DISK_USAGE_CRAWL_DELAY` set to 0, so far it does not look like that old files were deleted:

```
Name      : 178.png
Date      : 2020-11-14 03:37:21 UTC 
Size      : 2.1 KiB 
ETag      : 615c48adb3621139910c50a53aad85b7 
Type      : file 
Expiration: 2020-11-16 00:00:00 UTC (lifecycle-rule-id: delete after 1 day) 
Metadata  :
  Content-Type: application/octet-stream 
```

any way to debug this further?

**Comment by cubbasa** [comment]
@pew after upgrade to Minio at least RELEASE.2020-12-10T01-54-29Z I will continue my tests and probably know more and  get more results (currently I'm still using extra cron job to remove expired files). New version of Minio support options for "Dynamic systems without restarting server" (https://docs.min.io/docs/minio-server-configuration-guide.html) and (https://github.com/minio/minio/pull/11018). Without these options I can't modify options on the fly and see if they works because with my current version of Minio my setup after restart take a few hours to become fully operational again.

**Comment by harshavardhana** [comment]
> Now I've been running minio for a while with the `MINIO_DISK_USAGE_CRAWL_DELAY` set to 0, so far it does not look like that old files were deleted:
> 
> ```
> Name      : 178.png
> Date      : 2020-11-14 03:37:21 UTC 
> Size      : 2.1 KiB 
> ETag      : 615c48adb3621139910c50a53aad85b7 
> Type      : file 
> Expiration: 2020-11-16 00:00:00 UTC (lifecycle-rule-id: delete after 1 day) 
> Metadata  :
>   Content-Type: application/octet-stream 
> ```
> 
> any way to debug this further?

This means that lifecycle was never invoked.

**Comment by harshavardhana** [comment]
> options I can't modify options on the fly and see if they works because with my current version of Minio my setup after restart take a few hours to become fully operational again.

@cubbasa do you have like 1000's of buckets? this doesn't make sense. Unless you have ridiculously slow disks and your deployment is across some really slow slow network.

**Comment by pew** [comment]
> > Now I've been running minio for a while with the `MINIO_DISK_USAGE_CRAWL_DELAY` set to 0, so far it does not look like that old files were deleted:
> > ```
> > Name      : 178.png
> > Date      : 2020-11-14 03:37:21 UTC 
> > Size      : 2.1 KiB 
> > ETag      : 615c48adb3621139910c50a53aad85b7 
> > Type      : file 
> > Expiration: 2020-11-16 00:00:00 UTC (lifecycle-rule-id: delete after 1 day) 
> > Metadata  :
> >   Content-Type: application/octet-stream 
> > ```
> > 
> > 
> > any way to debug this further?
> 
> This means that lifecycle was never invoked.

Right, this is why I opened this issue since lifecycle policies don't seem to work correct in my environment.

**Comment by harshavardhana** [comment]
> Right, this is why I opened this issue since lifecycle policies don't seem to work correct in my environment.

Do you have a single drive mode? erasure coded single server? distributed erasure coded? @pew 

Can you enable `MINIO_DISK_USAGE_CRAWL_DEBUG=on`  and paste the logs after 24hrs?

**Comment by pew** [comment]
@harshavardhana  Single drive mode. `MINIO_DISK_USAGE_CRAWL_DEBUG=on` is already turned on, I put some requested information [already in this issue](https://github.com/minio/minio/issues/10980#issuecomment-737639869)

**Comment by harshavardhana** [comment]
@pew ah single drive interesting from the logs it looks like lifecycle is only invoked once. 

i.e 

>                          logger.Info(color.Green("crawlBucket:") + " lifecycle: Active rules found")

would you be able to deploy a binary that would help debug this?

**Comment by pew** [comment]
@harshavardhana I think so, I'm running the minio helm release in kubernetes but I can deploy that binary in the pod directly.

**Comment by harshavardhana** [comment]
> @harshavardhana I think so, I'm running the minio helm release in kubernetes but I can deploy that binary in the pod directly.

y4m4/minio:ilm - and keep the debug on @pew and please share after 5mins as the crawler starts running.

**Comment by pew** [comment]
@harshavardhana Thank you, I replaced the image and here's an excerpt of the log file:

```
dataUpdateTracker: Marking path dirty: bucketA/bee74c0f-c8e9-4eb6-97de-bd343288b681/8
dataUpdateTracker: Marking path dirty: bucketA/bee74c0f-c8e9-4eb6-97de-bd343288b681/8/136
dataUpdateTracker: Marking path dirty: bucketA/bee74c0f-c8e9-4eb6-97de-bd343288b681/8/136/165.png

API: SYSTEM()
Time: 06:10:09 UTC 12/13/2020
DeploymentID: 67f8e239-73ce-4b4b-865b-904e096d8d76
Error: cache version deprecated (will autoupdate) (*errors.errorString)
       3: cmd/data-usage-cache.go:485:cmd.(*dataUsageCache).load()
       2: cmd/fs-v1.go:257:cmd.(*FSObjects).CrawlAndGetDataUsage()
       1: cmd/data-crawler.go:112:cmd.runDataCrawler()
crawlBucket: lifecycle: Active rules found
data-usage: Start crawling. Bloom filter: false - /minio + bucketB
data-usage: Cycle: 0, Entries: 0 - /minio + bucketB
data-usage: Level 0, scanning 1 directories. - /minio + bucketB
folder-scanner: Prefix "" has active rules
data-usage: Level 1, scanning 150138 directories. - /minio + bucketB
folder-scanner: Prefix "f6f78b58-c914-4253-98be-613516727632" has active rules
folder-scanner: Prefix "e801d0ec-9d04-4786-bfa9-135a9b0e2528" has active rules
folder-scanner: Prefix "c7fd0f63-cddc-4a4f-a794-8a1ca9b58e9e" has active rules
folder-scanner: Prefix "e36ba78e-424d-4238-a8b4-3ab7e16d9a6a" has active rules
folder-scanner: Prefix "a3943c14-7e5d-4e50-bdc3-758ee25fafa4" has active rules
folder-scanner: Prefix "5a5e0a68-92ed-48b2-b6a2-584b691308b9" has active rules
folder-scanner: Prefix "741d25ab-dba4-4b52-b7b8-19ffa9adb5d6" has active rules
folder-scanner: Prefix "fd535bb9-1de2-477b-aa63-1ef9d0efd873" has active rules
folder-scanner: Prefix "fef3bb83-d084-4f14-b335-1084eb1332da" has active rules
folder-scanner: Prefix "a626b112-97a0-4b6b-b212-6ee3cf48fdf7" has active rules
folder-scanner: Prefix "f75bba63-1793-4f48-b00a-9784e34d90ec" has active rules
folder-scanner: Prefix "5bd3b8be-ce9b-4df5-91d7-a65feab101a3" has active rules
folder-scanner: Prefix "e1e75e7d-c255-408d-8e5b-4bf678f26f65" has active rules
folder-scanner: Prefix "7c164df5-b6f1-457e-aa98-45174a93aff1" has active rules
folder-scanner: Prefix "7731726f-0984-41ce-b846-c23ca896395f" has active rules
folder-scanner: Prefix "8dc9bb3f-8a77-4acd-a14c-06b92cec4673" has active rules
folder-scanner: Prefix "a0d91db5-7165-4f58-b295-b6e5f2eca10f" has active rules
folder-scanner: Prefix "d99bf698-310d-489c-a35a-ee10cf9cf7af" has active rules
folder-scanner: Prefix "c80efc0b-f8e5-4b30-9218-454c7daf4fbd" has active rules
folder-scanner: Prefix "b9aa5b39-2acc-4cbc-97b1-0c9acaa3f475" has active rules
```

the list goes on and on for all the individual prefixes across the buckets, I couldn't spot anything else.

**Comment by harshavardhana** [comment]
> the list goes on and on for all the individual prefixes across the buckets, I couldn't spot anything else.

This is very interesting so it indeed applies the rules but it's not able to see these objects as expirable for some reason, would you be able to copy a sample file to `play/testbucket` with `mc cp -a` and then provide the rules added by you on the same bucket? 

For me, it looks like there is a bug in the lifecycle evaluation crawler is actually doing its job.

**Comment by harshavardhana** [comment]
Don't you see any logs where it says `applyActions:` @pew ?

**Comment by pew** [comment]
> > the list goes on and on for all the individual prefixes across the buckets, I couldn't spot anything else.
> 
> This is very interesting so it indeed applies the rules but it's not able to see these objects as expirable for some reason, would you be able to copy a sample file to `play/testbucket` with `mc cp -a` and then provide the rules added by you on the same bucket?
> 
> For me, it looks like there is a bug in the lifecycle evaluation crawler is actually doing its job.

I just uploaded the folder I mentioned a few times in here already, the file I used as reference is now stored in `play/testbucket/8/142/178.png`.

This is the lifecycle json file I'm using:

```
{
	"Rules": [{
		"Expiration": {
			"Days": 1
		},
		"ID": "delete after 1 day",
		"Filter": {
			"Prefix": ""
		},
		"Status": "Enabled"
	}]
}
```

and this is `mc ilm export`

```
{
 "Rules": [
  {
   "AbortIncompleteMultipartUpload": {},
   "Expiration": {
    "Date": "0001-01-01T00:00:00Z",
    "Days": 1,
    "DeleteMarker": false
   },
   "ID": "delete after 1 day",
   "Filter": {
    "And": {}
   },
   "NoncurrentVersionExpiration": {
    "NoncurrentDays": 0
   },
   "NoncurrentVersionTransition": {},
   "Status": "Enabled",
   "Transition": {
    "Date": "0001-01-01T00:00:00Z"
   }
  }
 ]
}
```

also, I was just searching the whole log file since starting the server for `applyActions` and so far there are no results.

**Comment by harshavardhana** [comment]
> > > the list goes on and on for all the individual prefixes across the buckets, I couldn't spot anything else.
> > 
> > 
> > This is very interesting so it indeed applies the rules but it's not able to see these objects as expirable for some reason, would you be able to copy a sample file to `play/testbucket` with `mc cp -a` and then provide the rules added by you on the same bucket?
> > For me, it looks like there is a bug in the lifecycle evaluation crawler is actually doing its job.
> 
> I just uploaded the folder I mentioned a few times in here already, the file I used as reference is now stored in `play/testbucket/8/142/178.png`.
> 
> This is the lifecycle json file I'm using:
> 
> ```
> {
> 	"Rules": [{
> 		"Expiration": {
> 			"Days": 1
> 		},
> 		"ID": "delete after 1 day",
> 		"Filter": {
> 			"Prefix": ""
> 		},
> 		"Status": "Enabled"
> 	}]
> }
> ```
> 
> and this is `mc ilm export`
> 
> ```
> {
>  "Rules": [
>   {
>    "AbortIncompleteMultipartUpload": {},
>    "Expiration": {
>     "Date": "0001-01-01T00:00:00Z",
>     "Days": 1,
>     "DeleteMarker": false
>    },
>    "ID": "delete after 1 day",
>    "Filter": {
>     "And": {}
>    },
>    "NoncurrentVersionExpiration": {
>     "NoncurrentDays": 0
>    },
>    "NoncurrentVersionTransition": {},
>    "Status": "Enabled",
>    "Transition": {
>     "Date": "0001-01-01T00:00:00Z"
>    }
>   }
>  ]
> }
> ```
> 
> also, I was just searching the whole log file since starting the server for `applyActions` and so far there are no results.

Yeah that is super weird since it says it has active rules but at the same time there are no logs related to apply actions.

Thanks I will get back I have enough information to look for.

**Comment by harshavardhana** [comment]
> also, I was just searching the whole log file since starting the server for `applyActions` and so far there are no results.

How many objects do you have per bucket?

**Comment by harshavardhana** [comment]
@pew one more container for you to deploy with some more additional messages for us to check for.  For the same image feel free to re-pull and deploy. 

```
docker pull y4m4/minio:ilm
```

**Comment by pew** [comment]
> > also, I was just searching the whole log file since starting the server for `applyActions` and so far there are no results.
> 
> How many objects do you have per bucket?

I'd say hundreds of thousand of these very small-ish files, across three buckets. The lifecycle policy of one day should only apply to one buckets, the other ones keep files a little longer.

I re-deployed the image, let me know if you need any specific messages, so far the output looks like the same as before.

**Comment by harshavardhana** [comment]
> > > also, I was just searching the whole log file since starting the server for `applyActions` and so far there are no results.
> > 
> > 
> > How many objects do you have per bucket?
> 
> I'd say hundreds of thousand of these very small-ish files, across three buckets. The lifecycle policy of one day should only apply to one buckets, the other ones keep files a little longer.
> 
> I re-deployed the image, let me know if you need any specific messages, so far the output looks like the same as before.

We need to see anything with `applyActions:`  @pew as a prefix for messages.

**Comment by pew** [comment]
@harshavardhana this is what I got so far:

```
applyActions: lifecycle: "e36ba78e-424d-4238-a8b4-3ab7e16d9a6a/8/128/171.png" Initial scan: DeleteAction
applyActions: lifecycle: Secondary scan: DeleteAction
applyActions: lifecycle: "e36ba78e-424d-4238-a8b4-3ab7e16d9a6a/8/128/172.png" Initial scan: DeleteAction
applyActions: lifecycle: Secondary scan: DeleteAction
applyActions: lifecycle: "e36ba78e-424d-4238-a8b4-3ab7e16d9a6a/8/128/173.png" Initial scan: DeleteAction
applyActions: lifecycle: Secondary scan: DeleteAction
applyActions: lifecycle: "e36ba78e-424d-4238-a8b4-3ab7e16d9a6a/8/128/174.png" Initial scan: DeleteAction
applyActions: lifecycle: Secondary scan: DeleteAction
applyActions: lifecycle: "e36ba78e-424d-4238-a8b4-3ab7e16d9a6a/8/128/175.png" Initial scan: DeleteAction
applyActions: lifecycle: Secondary scan: DeleteAction
applyActions: lifecycle: "e36ba78e-424d-4238-a8b4-3ab7e16d9a6a/8/128/176.png" Initial scan: DeleteAction
applyActions: lifecycle: Secondary scan: DeleteAction
applyActions: lifecycle: "e36ba78e-424d-4238-a8b4-3ab7e16d9a6a/8/128/177.png" Initial scan: DeleteAction
applyActions: lifecycle: Secondary scan: DeleteAction
applyActions: lifecycle: "e36ba78e-424d-4238-a8b4-3ab7e16d9a6a/8/130/178.png" Initial scan: DeleteAction
```

the list goes on. When I do a `mc stat` on these files they're not there. Also the one `178.png` I mentioned in this thread is gone. It does seem to work... now? I hope there are some other changes you made to this image, since the one file was there for weeks now. :)

**Comment by harshavardhana** [comment]
Yes I did @pew it's in the PR #11096

**Comment by harshavardhana** [comment]
fixed by #11096
